google_photos_takeout_helper/__main__.py

```diff
@@ -483,41 +524,44 @@ def fix_metadata(file: Path):
             has_nice_date = True
             return
         except FileNotFoundError:
-            print("Couldn't find json for file :/")
+            print("Couldn't find json for file ")
 
         if has_nice_date:
-            return
+            return True
 
-        print('Last chance, coping folder name as date...')
-        date = get_date_from_folder_name(file.parent)
-        set_file_exif_date(file, date)
-        set_creation_date_from_str(file, date)
-
-        nonlocal s_date_from_folder_files
-        s_date_from_folder_files.append(str(file.resolve()))
+        print('Last chance, coping folder meta as date...')
+        date = get_date_from_folder_meta(file.parent)
+        if date is not None:
+            set_file_exif_date(file, date)
+            set_creation_date_from_str(file, date)
+            nonlocal s_date_from_folder_files
+            s_date_from_folder_files.append(str(file.resolve()))
+            return True
+        else:
+            print('ERROR! There was literally no option to set date!!!')
+            # TODO
+            print('TODO: We should do something about this - move it to some separate folder, or write it down in '
+                  'another .txt file...')
```

Comment:
In such tragic scenario, I think we should do something different - I think the .txt and BIG disclaimer at the end will be enough